RELEASE.md

```diff
@@ -0,0 +1,139 @@
+# web3.js Release Guidelines
+
+## Version Number Definition
+
+The web3.js project follows the [semver 2.0.0 specification](https://semver.org/).
+
+### Major
+
+The `major` version has to be increased as soon as a breaking change is introduced. The definition of a breaking change is anything that requires a depending project to update their code base, build pipeline or tests.
+
+### Minor
+
+The `minor` is increased as soon as new smaller features do get introduced. A `minor` release will not affect any depending project. Such a release only introduces smaller enhancements and features a project could use.
+
+### Patch
+
+A patch release only contains required bug fixes with a low risk to impact depending project.
+
+Further details about versioning can be found in the [semver 2.0.0 specification](https://semver.org/) we follow.
+
+## Release Process Rules
+
+### Major
+
+1.  The release should get released as an `RC` version on the specified track below.
+1.  The minor `RC` version stays released for a `month` for testing purposes.
+1.  During release review, the code is frozen unless new changes are proposed, approved and merged.
+    1.  Changes should trigger a new `RC` release and set the release clock back a `week` so reviewers have the time they need to test new changes.
+1.  The project lead should manually test the `RC` against a Node project and the published
+    minified bundle in a browser context. An external reviewer should verify they have done the same.
+
+### Minor
+
+The `minor` release inherits the rules from the `major` release but has a smaller testing phase of one `week` and the release clock will be set back `2 days` on a change of the published code.
+
+### Patch
+
+Since November 2019, Web3 has been following the `minor` rules for `patch` as well, to help establish a series of non-breaking releases.
+
+## Prereleases
+
+### Release Candidate
+
+A release candidate gets published on the `rc` track and is no longer under active development. A release candidate gets released with the version number defined in the [semver 2.0.0 specification](https://semver.org/) specification.
+
+### Beta Release
+
+A `beta` release gets published on the `beta` track and is under active developemnt. The version number of the `beta` release gets defined as specified in the [semver 2.0.0](https://semver.org/) document.
+
+## Long Term Support (LTS)
+
+> Note: This section has not yet been decided and so for now is purely informative.
+
+To provide safety over longer periods, it is important to have versions tagged as `LTS`.
+
+| Release |    Status     | Initial Release |   LTS Start   | End-of-Life |
+| :-----: | :-----------: | :-------------: | :-----------: | :---------: |
+|   1.x   |      LTS      |  24. Jul. 2017  | 23. Jul. 2019 |     TBD     |
+|   2.x   | Alpha Preview |  13. Jul. 2019  |      TBD      |     TBD     |
```

Comment:
We discussed the LTS and EOL of 2.x already with Embark, Travis, etc... If I'm correct did we said 2.0 stable release plus 12months == EOL 1.0